src/aus.h

```diff
@@ -0,0 +1,305 @@
+
+        long get()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            if (_is_running)
+            {
+                return get_current_time() - _start + _total;
+            }
+            else
+            {
+                return _total;
+            }
+        }
+        
+        void set(long value) {
+            return;
+        }
+
+        void increment()
+        {
+            return;
+        }
+
+        void decrement()
+        {
+            return;
+        }
+
+        void start()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            _start = std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+            _is_running = true;
+        }
+
+        void stop()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            if (!_is_running)
+            {
+                return;
+            }
+            _is_running = false;
+            long current_time = get_current_time();
+            _total += (current_time - _start);
+            _start = current_time;
+        }
+
+
+
+    private:
+        long _start;
+        long _total;
+        bool _is_running;
+        std::mutex _m;
+
+
+        time_t get_current_time() 
+        {
+            return std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+        }
+
+    };
+
+    class aus_counter : public aus_value
+    {
+    public:
+        aus_counter(long value = 0) : _counter(value), _total(0) { }
+        
+        long get()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            return _counter;
+        }
+
+        void set(long value) {
+            std::lock_guard<std::mutex> lock(_m);
+            _counter = value;
+        }
+
+        void increment()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            _counter++;
+        }
+
+        void decrement()
+        {
+            std::lock_guard<std::mutex> lock(_m);
+            _counter--;
+        }
+
+        void start() { return; }
+
+        void stop() { return; }
+
+
+    private:
+        long _counter;
+        long _total;
+        std::mutex _m;
+    };
+
+    class aus_data
+    {
+
+    public:
+        aus_data()
+        {
+            _start_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+            _librealsense_version = RS2_API_VERSION_STR;
+            _os_name = get_os_name();
+            _platform_name = PLATFORM;
+
+        }
+
+        std::string get_os_name()
+        {
+            #ifdef _WIN32
+            return "Windows";
+            #else
+            #ifdef __APPLE__
+            return "Mac OS";
+            #else
+            #ifdef __linux__
+            return "Linux";
+            #else
+            return "Unknown";
+            #endif
+            #endif
+            #endif
+        }
+
+        
+        void set(std::string key, long value)
```

Comment:
AUS data class should be protected with mutex as well,

Reply:
added mutex to aus_data class and removed the mutex from aus_counter and aus_timer